kcqrs-core/src/main/kotlin/com/clouway/kcqrs/core/SimpleMessageBus.kt

```diff
@@ -25,11 +25,12 @@ class SimpleMessageBus : MessageBus {
 
 
     @SuppressWarnings("unchecked")
-    override fun <T : Command> registerCommandHandler(aClass: Class<T>, handler: CommandHandler<T>) {
+    override fun <T : Command> registerCommandHandler(aClass: Class<T>, handler: CommandHandler<T>, validation: Validation<T>) {
```

Comment:
Maybe the old one could be left so that we could register command handlers without validation as well. A default validation returning always true could be provided underline.

Reply:
This is how it's done currently. 

Check the definition of the MessageBus interface: 

```kotlin
    fun <T : Command> registerCommandHandler(aClass: Class<T>, handler: CommandHandler<T>, validation: Validation<T> = Validation {})
```